pkg/stargate/vector_test.go

```diff
@@ -0,0 +1,36 @@
+package stargate
+
+import (
+	"testing"
+
+	testlogr "github.com/go-logr/logr/testing"
+	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
+	api "github.com/k8ssandra/k8ssandra-operator/apis/stargate/v1alpha1"
+	telemetryapi "github.com/k8ssandra/k8ssandra-operator/apis/telemetry/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/vector"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/utils/pointer"
+)
+
+// InjectCassandraVectorAgent adds the Vector agent container to the Cassandra pods.
+// If the Vector agent is already present, it is not added again.
+func TestConfigureStargate(t *testing.T) {
```

Comment:
Nit: the test name should rather be something like `TestConfigureVector`. Also, the comment above seems unrelated.